nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -367,8 +377,35 @@ private static boolean isDirectTypeUseAnnotation(Attribute.TypeCompound t, Confi
           return false;
       }
     }
-    // Make sure it's not a mix of inner types and arrays for this annotation's location
-    return !(locationHasInnerTypes && locationHasArray);
+    if (config.isLegacyAnnotationLocation()) {
+      // Make sure it's not a mix of inner types and arrays for this annotation's location
+      return !(locationHasInnerTypes && locationHasArray);
+    }
+    // For non-nested classes if there are any inner types in the annotation location the annotation
+    // is considered valid to allow annotations on type arguments.
+    if (!hasNestedClass(symbol.type)) {
+      if (innerTypeCount > 0) {
+        return true;
+      }
+    }
+    // For nested classes the annotation is only valid if it is on the innermost type.
+    return innerTypeCount == nestingDepth - 1;
```

Comment:
This code also executes for non-nested classes if `innerTypeCount == 0`.  It seems that is expected?  In any case the comment needs to be clarified

Reply:
Fixed!